### PR TITLE
Refactor/142 thruster platform reference integral feedback

### LINKED ISSUE #142: thrusterPlatformReference refactor
**Describe your use case**
This issue aims to add an integral feedback to the control law in thrusterPlatformReference that uses the thruster platform to dump momentum on the reaction wheels. Additionally, a user input is added to provide a limit to the gimbal angles allowed.

**Describe alternatives solutions you've considered**
An alternative solution consists in using only the proportional feedback term combined with continuous center of mass (CoM) estimation.

**Additional context**
When the CoM is not continuously estimated, the bias on the CoM knowledge causes a steady-state bias on the wheel speeds, which cannot be dumped by the control law. Adding the integral feedback term aims to dump that steady-state bias.

### PR DESCRIPTION
**Pull request by rcalaon**
* **Tickets addressed:** bsk-142
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
An integral feedback term is provided as an input parameter. When the input is negative, or not provided, the integral feedback term is not added, and the control law defaults to a proportional type. 

Two input parameters are define to bound the absolute value of the reference angles. When the computed references exceed these bounds, they are set equal to the bounds. When the inputs are negative, the bounding of the reference angles is bypassed.

Commit 1 addresses an issue in the former version of the module, making sure that the momentum dumping routine is run only when the RW speed message is connected. Commit 2 implements the bounding of the reference angles. Commit 3 computes the integral feedback term and adds it to the control law. Commits 4 and 5 update the unit test and documentation, respectively.

## Verification
A unit test is provided to test the correctness of the output, reflecting the most recent changes.

## Documentation
Documentation is updated to reflect the changes.

## Future work
No future work is foreseen at the moment.

### CODE REVIEW

**Review by codyallard28 — COMMENTED** [review]
@rcalaon, can you add the integral term to your scenario since it is isn't being testing in the unit test?

**Comment by rcalaon** [comment]
> @rcalaon, can you add the integral term to your scenario since it is isn't being testing in the unit test?

There's a branch analyses/120-momentumManagementAnalysis that already contains the integral term

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c]: :large_orange_diamond: Variables should each be declared in on their own new line.

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c]: :large_orange_diamond: Is there a case in which this would ever not be a 3 vector? If not, would `v3SetZero` work well here?

**Review by patkenneally — APPROVED** [review]

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c
Comment (patkenneally) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c]: Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Comment (rcalaon) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c]: My bad, I was copying some legacy code here. I fixed it.

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c
Comment (rcalaon) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c]: Absolutely. I'd forgotten we had that function. Fixed it.

**Pull request merged**
